### PR TITLE
Issue/2639 form export full (connect #2639)

### PR DESCRIPTION
**Pull request by stellanl**
#### Before the PR (what is the issue or what needed to be done)
Form export was just meant for a paper form
#### The solution
Add another sheet that shows details for all groups and questions of entire form as a table.
## Checklist
* [x] Connect the issue
* [ ] Test plan
* [ ] Copyright header
* [ ] Code formatting
* [ ] Documentation

### CODE REVIEW

**Comment by valllllll2000** [comment]
Please update copyright of files and check codacy errors

File: GAE/src/org/waterforpeople/mapping/dataexport/SurveyFormExporter.java [hunk 1]
```diff
@@ -196,10 +221,174 @@ private void writeSurvey(String title, File fileName,
...
+                sheet.addMergedRegion(new CellRangeAddress(firstRowInGroup, startRow + count - 1, 1, 1));
+                sheet.addMergedRegion(new CellRangeAddress(firstRowInGroup, startRow + count - 1, 2, 2));
+            }
+        }
+    }
+
+    private int createFullHeader(HSSFSheet sheet, int startRow, HSSFCellStyle style) {
+        int r = startRow, c = 0;
+        HSSFRow row = sheet.createRow(r++);
+        //createCell(row, c++, "Form version", style);
```
Comment (valllllll2000) [inline GAE/src/org/waterforpeople/mapping/dataexport/SurveyFormExporter.java]: Shall we maybe remove the commented out code?

File: GAE/src/org/waterforpeople/mapping/dataexport/SurveyFormExporter.java [hunk 2]
```diff
@@ -217,11 +406,117 @@ private HSSFCell createCell(HSSFRow row, int col, String value,
...
+            cell.setCellStyle(style);
+        }
+        if (value != null && value) {
+            cell.setCellValue("Yes");
+        }
+        return cell;
+    }
+
+    /**
+     * creates a cell in the row passed in and sets the style and value (if
```
Comment (valllllll2000) [inline GAE/src/org/waterforpeople/mapping/dataexport/SurveyFormExporter.java]: I think, the comment should explain more what it does with width

File: GAE/src/org/waterforpeople/mapping/dataexport/SurveyFormExporter.java [hunk 3]
```diff
@@ -196,10 +229,187 @@ private void writeSurvey(String title, File fileName,
...
+
+        row = sheet.createRow(r++);
+        createCellBlock(row, 0, "Group", style, 3);
+        createCellBlock(row, 3, "Question", style, 3);
+        c = 12; //Group headers
+        createCellBlock(row, c, "Dependency", style, 3); // 3 wide
+        createCellBlock(row, c += 3, "Numbers", style, 4); // 4 wide
+        createCellBlock(row, c += 4, "Options", style, 3); // 3 wide
+        createCellBlock(row, c += 3, "Geolocation", style, 2); // 2 wide
+        createCell(     row, c += 2, "Cascade", style);
```
Comment (valllllll2000) [inline GAE/src/org/waterforpeople/mapping/dataexport/SurveyFormExporter.java]: isn't there too much spacing here after '('

File: GAE/src/org/waterforpeople/mapping/dataexport/SurveyFormExporter.java [hunk 4]
```diff
@@ -196,10 +229,187 @@ private void writeSurvey(String title, File fileName,
...
+        createCellBlock(row, c += 1, "Geographic area", style, 3); // 3 wide
+        createCell(     row, c += 3, "Caddisfly", style);
```
Comment (valllllll2000) [inline GAE/src/org/waterforpeople/mapping/dataexport/SurveyFormExporter.java]: same comment as above

File: GAE/src/org/waterforpeople/mapping/dataexport/SurveyFormExporter.java (hunk 3 above)
Comment (stellanl) [reply GAE/src/org/waterforpeople/mapping/dataexport/SurveyFormExporter.java]: I wanted it to align vertically so you can see the pattern of it.

**Pull request merged**